server/service/confluence_service.go

```diff
@@ -0,0 +1,198 @@
+package service
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"path"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+const ErrorStatusNotFound = "No content found"
+
+type ConfluenceStatus struct {
+	State string `json:"state"`
+}
+
+type ErrorResponse struct {
+	Message string `json:"message"`
+}
+
+func NormalizeConfluenceURL(confluenceURL string) (string, error) {
+	u, err := url.Parse(confluenceURL)
+	if err != nil {
+		return "", err
+	}
+
+	if u.Host == "" {
+		ss := strings.Split(u.Path, "/")
+		if len(ss) > 0 && ss[0] != "" {
+			u.Host = ss[0]
+			u.Path = path.Join(ss[1:]...)
+		}
+		u, err = url.Parse(u.String())
+		if err != nil {
+			return "", err
+		}
+	}
+
+	if u.Host == "" {
+		return "", errors.Errorf("Invalid URL, no hostname: %q", confluenceURL)
+	}
+	if u.Scheme == "" {
+		u.Scheme = "https"
+	}
+
+	confluenceURL = strings.TrimSuffix(u.String(), "/")
+	return confluenceURL, nil
+}
+
+// CheckConfluenceURL checks if the `/status` endpoint of the Confluence URL is accessible
+// and responding with the correct state which is "RUNNING"
+func CheckConfluenceURL(mattermostSiteURL, confluenceURL string, requireHTTPS bool) (_ string, err error) {
+	confluenceURL, err = NormalizeConfluenceURL(confluenceURL)
+	if err != nil {
+		return "", err
+	}
+
+	if confluenceURL == strings.TrimSuffix(mattermostSiteURL, "/") {
+		return "", errors.Errorf("%s is the Mattermost site URL. Please use your Confluence URL", confluenceURL)
```

Comment:
😅

Reply:
@fmartingr I think this check seems extra here 😅, but it's harmless. Should we remove it?

Reply:
I don't think is necessary, did you add this because you saw this happening at some point?